linode/objbucket/resource.go

```diff
@@ -83,13 +84,23 @@ func readResource(
 			return diag.Errorf("access_key and secret_key are required to get versioning and lifecycle info")
 		}
 
-		conn := helper.S3ConnFromResourceData(d)
+		if endpoint == "" {
+			var err error
+			if endpoint, err = helper.ComputeS3Endpoint(ctx, d, meta); err != nil {
+				return diag.FromErr(err)
+			}
+		}
```

Comment:
We don't need to compute the endpoint here since it's already available in the `bucket` object above.

e.g.
```go
endpoint := strings.TrimPrefix(bucket.Hostname, fmt.Sprintf("%s.", bucket.Label))
```

It might also make sense to move this out of the if block to make sure `endpoint` is always populated :+1:

Reply:
great feedback, ill change it